src/types/index.ts

```diff
@@ -1,17 +1,37 @@
 export interface PostResponse {
-  id: number;
+  id: string;
+  date: string;
   title: {
     rendered: string;
   };
   content: {
     rendered: string;
   };
+  excerpt: {
+    rendered: string;
+  };
+  featured_media: string;
 }
 
 export interface Post {
-  id: number;
+  id: string;
+  date: string;
   title: string;
   content: string;
+  excerpt: string;
+  featured_media: string;
+}
+
+export interface ImageResponse {
```

Comment:
Are those API models or Application models? I think it would be good to separate these and also use only the needed values -> proper mappers `ApiModel->ApplicationModel`